test/test_kfac.py

```diff
@@ -519,3 +520,200 @@ def test_torch_matvec(case):
+    for aaT in kfac._input_covariances.values():
+        aaT.add_(
+            torch_eye(aaT.shape[0], dtype=aaT.dtype, device=aaT.device), alpha=delta
+        )
+    for ggT in kfac._gradient_covariances.values():
+        ggT.add_(
+            torch_eye(ggT.shape[0], dtype=ggT.dtype, device=ggT.device), alpha=delta
+        )
+
+    # Check for equivalence of the det property and naive determinant computation
+    determinant = kfac.det
+    # verify that the determinant is not trivial as this would make the test useless
+    assert determinant != 0.0 and determinant != 1.0
+    det_naive = det(kfac @ eye(kfac.shape[1]))
+    report_nonclose(determinant.cpu().numpy(), det_naive)
+
+    # Check that the det property is properly cached and reset
+    assert kfac._det == determinant
+    kfac._compute_kfac()
+    assert kfac._det is None
+
+
+@mark.parametrize(
+    "check_deterministic",
+    [True, False],
+    ids=["check_deterministic", "dont_check_deterministic"],
+)
+@mark.parametrize(
+    "separate_weight_and_bias", [True, False], ids=["separate_bias", "joint_bias"]
+)
+@mark.parametrize(
+    "exclude", [None, "weight", "bias"], ids=["all", "no_weights", "no_biases"]
+)
+def test_logdet(case, exclude, separate_weight_and_bias, check_deterministic):
+    """Test that the log determinant property of KFACLinearOperator works."""
+    model, loss_func, params, data = case
+
+    if exclude is not None:
+        names = {p.data_ptr(): name for name, p in model.named_parameters()}
+        params = [p for p in params if exclude not in names[p.data_ptr()]]
+
+    loss_average = None if loss_func.reduction == "sum" else "batch"
+    kfac = KFACLinearOperator(
+        model,
+        loss_func,
+        params,
+        data,
+        separate_weight_and_bias=separate_weight_and_bias,
+        loss_average=loss_average,
+        check_deterministic=check_deterministic,
+    )
+
+    # add damping manually to avoid singular matrices
+    if not check_deterministic:
+        kfac._compute_kfac()
+    assert kfac._input_covariances or kfac._gradient_covariances
+    delta = 1e-7  # only requires much smaller damping value compared to ``det``
+    for aaT in kfac._input_covariances.values():
+        aaT.add_(
+            torch_eye(aaT.shape[0], dtype=aaT.dtype, device=aaT.device), alpha=delta
+        )
+    for ggT in kfac._gradient_covariances.values():
+        ggT.add_(
+            torch_eye(ggT.shape[0], dtype=ggT.dtype, device=ggT.device), alpha=delta
+        )
+
+    # Check for equivalence of the logdet property and naive log determinant computation
+    log_det = kfac.logdet
+    # verify that the log determinant is finite and not nan
+    assert not isinf(log_det) and not isnan(log_det)
+    sign, logabsdet = slogdet(kfac @ eye(kfac.shape[1]))
+    det_through_logdet_naive = sign * exp(logabsdet)
+    report_nonclose(exp(log_det.cpu().numpy()), det_through_logdet_naive)
```

Comment:
Q: Why not compare `log_det` with `sign * logabsdet`?

Reply:
Was numerically less similar, but I just increased the damping and now compare the log dets directly.